src/main/java/uk/gov/pay/ledger/transaction/service/TransactionService.java

```diff
@@ -100,6 +100,15 @@ private TransactionSearchResponse searchTransactionsWithoutParent(TransactionSea
 
         Long total = transactionDao.getTotalForSearch(searchParams);
 
+        long size = searchParams.getDisplaySize();
+        if (total > 0 && searchParams.getDisplaySize() > 0) {
+            long lastPage = (total + size - 1) / size;
+            if (searchParams.getPageNumber() > lastPage || searchParams.getPageNumber() < 1) {
```

Comment:
Shouldn't we have validation of `searchParams.getPageNumber() < 1` before we run any db queries?
(I'm trying to prematurely optimise ;) )

Reply:
👍 Will look into it separately